### PR TITLE
feat: add support to create all directories under RAD_HOME

### PR DESCRIPTION
**Pull request by MeBrei**
Solves part of #1199 

When setting the env variable `RAD_HOME`, the user can decide to store all directories created by upstream and link under the given directory. This is already supported in link it self.
This can also be used to create an isolated dev environment.

It is separated out of the multi-identity work, as this part is a non-breaking change.

Signed-off-by: Merle Breitkreuz <merle@monadic.xyz>

### CODE REVIEW

**Review by FintanH — APPROVED** [review]
Looks good :+1:

File: proxy/api/src/config.rs [hunk 1]
```diff
@@ -27,6 +27,18 @@ pub fn dirs() -> ProjectDirs {
     ProjectDirs::from("xyz", "radicle", "radicle-upstream").expect("couldn't build dirs")
 }
 
+/// Returns the directory for the application store
+pub fn store_dir() -> path::PathBuf {
+    let store_root = match std::env::var_os("RAD_HOME") {
+        None => {
+            let dirs = dirs();
+            return dirs.data_dir().to_path_buf();
```
Comment (geigerzaehler) [inline proxy/api/src/config.rs:35]: Shouldn’t we also join this with a `store` segment?

File: native/main.ts [hunk 2]
```diff
@@ -32,6 +33,13 @@ if (process.env.RADICLE_UPSTREAM_PROXY_ARGS) {
   );
 }
 
+if (process.env.RAD_HOME) {
```
Comment (geigerzaehler) [inline native/main.ts:36]: Can we add some documentation for this in the developer docs?

**Review by kim — APPROVED** [review]

File: proxy/api/src/config.rs (hunk 1 above)
Comment (MeBrei) [reply proxy/api/src/config.rs:35]: `store` is joined at the end in both cases in [l.39](https://github.com/radicle-dev/radicle-upstream/pull/1527/files/0c8d50f0b5aef28081168663b39bc8501e4bdd6d#diff-c829644a81bfa71911f54a78e6369e3d4279dcb077586f7b47d92374213d6aeeR39)

Comment (geigerzaehler) [reply proxy/api/src/config.rs:35]: Are you sure? We `return` here.

Comment (FintanH) [reply proxy/api/src/config.rs:35]: Oh nice catch :smile:

Comment (MeBrei) [reply proxy/api/src/config.rs:35]: 🤦 👍

**Review by rudolfs — APPROVED** [review]

**Review by FintanH — APPROVED** [review]

**Review by geigerzaehler — APPROVED** [review]

**Comment by MeBrei** [comment]
Merged via 8bf3152453c1b3bdee98700264e7b37040a52569

**Comment by NunoAlexandre** [comment]
Edit: Merged in c04dc4b24eff0da1b32addc76170e5dd56fb6bd2

**Pull request closed without merging**